hashtable.c

```diff
@@ -0,0 +1,133 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "hashtable.h"
+#define TABLE_SIZE 1000
+
+typedef struct Node {
+	const char* word;
+	int count;
+	Node* next;
+} Node;
+
+typedef struct HashTable {
+	Node* table[TABLE_SIZE];
+} HashTable;
+
+Node* create_node(const char* word) {
+	Node* new_node = malloc(sizeof(Node));
+	new_node->word = word;
+	new_node->count = 1;
+	new_node->next = NULL;
+	return new_node;
+}
+
+void insert(Node** head, const char* word) {
+	Node* current = *head;
+	while (current != NULL) {
+		if (strcmp(current->word, word) == 0) {
+			current->count++;
+			return;
+		}
+		current = current->next;
+	}
+
+	Node* new_node = create_node(word);
+	if (*head != NULL) {
+		new_node->next = *head;
+		*head = new_node;
+	}
+	else {
+		*head = new_node;
+	}

```

Comment:
```suggestion
	if (*head != NULL) {
		new_node->next = *head;
	}
	*head = new_node;
```
Больше того, просто `*head = new_node;`. От того, что new_node->next присвоится NULL, хуже никому не станет.